samples/attested_tls/common/verify_callback.cpp

```diff
@@ -228,47 +230,75 @@ int verify_callback(int preverify_ok, X509_STORE_CTX* ctx)
     }
 
     // convert a cert into a buffer in DER format
-    der_len = i2d_X509(crt, nullptr);
-    buff = (unsigned char*)malloc((size_t)der_len);
-    if (buff == nullptr)
+    certificate_size = i2d_X509(crt, nullptr);
+    buffer = (unsigned char*)malloc((size_t)certificate_size);
+    if (buffer == nullptr)
     {
-        printf(TLS_CLIENT "malloc failed (der_len=%d)\n", der_len);
+        printf(
+            TLS_CLIENT "malloc failed (certificate_size=%d)\n",
+            certificate_size);
         goto done;
     }
-    der = buff;
-    der_len = i2d_X509(crt, &buff);
-    if (der_len < 0)
+    certificate = buffer;
+    certificate_size = i2d_X509(crt, &buffer);
+    if (certificate_size < 0)
     {
-        printf(TLS_CLIENT "i2d_X509 failed(der_len=%d)\n", der_len);
+        printf(
+            TLS_CLIENT "i2d_X509 failed(certificate_size=%d)\n",
+            certificate_size);
         goto done;
     }
 
     // note: i2d_X509() updates the pointer to the buffer so that following the
-    // call to i2d_X509(), buff is pointing to the "end" of the data buffer
-    // pointed by buff That is, buff = buff + der_len;
+    // call to i2d_X509(), buffer is pointing to the "end" of the data buffer
+    // pointed by buffer That is, buffer = buffer + certificate_size;
     printf(
-        TLS_CLIENT "der=%p buff=%p buff moved by %d offset der_len=%d\n",
-        der,
-        buff,
-        (int)(buff - der),
-        der_len);
+        TLS_CLIENT "certificate=%p buffer=%p buffer moved by %d offset "
+                   "certificate_size=%d\n",
+        certificate,
+        buffer,
+        (int)(buffer - certificate),
+        certificate_size);
 
     printf(" verifying certificate start \n");
     // verify tls certificate
     oe_verifier_initialize();
-    result = oe_verify_attestation_certificate_with_evidence(
-        der, (size_t)der_len, enclave_claims_verifier, nullptr);
+    result = oe_verify_attestation_certificate_with_evidence_v2(
+        certificate,
+        (size_t)certificate_size,
+        nullptr,
+        0,
+        nullptr,
+        0,
+        &claims,
+        &claims_length);
+
+    if (result != OE_OK)
+    {
+        printf(
+            TLS_CLIENT "oe_verify_attestation_certificate_with_evidence_v2 "
+                       "failed with result=%s\n",
+            oe_result_str(result));
+        goto done;
+    }
+
+    result = enclave_claims_verifier(claims, claims_length, nullptr);
+
     if (result != OE_OK)
     {
-        printf(TLS_CLIENT "result=%s\n", oe_result_str(result));
+        printf(
+            TLS_CLIENT "enclave_claims_verifier failed with result=%s\n",
+            oe_result_str(result));
         goto done;
     }
+
     printf(" verifying certificate end\n");
     ret = 1;
 done:
 
-    if (der)
-        free(der);
+    if (certificate)
+        free(certificate);
```

Comment:
```suggestion
    free(certificate);
```
`free()` should be safe to handle a `NULL` pointer.